warehouse/bigquery/bigquery.go

```diff
@@ -462,21 +495,212 @@ func (bq *HandleT) Cleanup() {
 }
 
 func (bq *HandleT) LoadIdentityMergeRulesTable() (err error) {
+	identityMergeRulesTable := warehouseutils.IdentityMergeRulesWarehouseTableName(PROVIDER)
+	pkgLogger.Infof("BQ: Starting load for table:%s\n", identityMergeRulesTable)
+
+	pkgLogger.Infof("BQ: Fetching load file location for %s", identityMergeRulesTable)
+
+	var location string
+	location, err = bq.Uploader.GetSingleLoadFileLocation(identityMergeRulesTable)
+	if err != nil {
+		return err
+	}
+	location = warehouseutils.GetGCSLocation(location, warehouseutils.GCSLocationOptionsT{})
+	pkgLogger.Infof("BQ: Loading data into table: %s in bigquery dataset: %s in project: %s from %v", identityMergeRulesTable, bq.Namespace, bq.ProjectID, location)
+	gcsRef := bigquery.NewGCSReference(location)
+	gcsRef.SourceFormat = bigquery.JSON
+	gcsRef.MaxBadRecords = 0
+	gcsRef.IgnoreUnknownValues = false
+	partitionDate := time.Now().Format("2006-01-02")
+	outputTable := partitionedTable(identityMergeRulesTable, partitionDate)
+
+	// create partitioned table in format tableName$20191221
+	loader := bq.Db.Dataset(bq.Namespace).Table(outputTable).LoaderFrom(gcsRef)
+
+	job, err := loader.Run(bq.BQContext)
+	if err != nil {
+		pkgLogger.Errorf("BQ: Error initiating load job: %v\n", err)
+		return
+	}
+	status, err := job.Wait(bq.BQContext)
+	if err != nil {
+		pkgLogger.Errorf("BQ: Error running load job: %v\n", err)
+		return
+	}
+
+	if status.Err() != nil {
+		return status.Err()
+	}
 	return
 }
 
 func (bq *HandleT) LoadIdentityMappingsTable() (err error) {
+	identityMappingsTable := warehouseutils.IdentityMappingsWarehouseTableName(PROVIDER)
```

Comment:
merge both LoadIdentityMergeRulesTable and LoadIdentityMappingsTable into one func. Or maybe reuse the loadTable func as we are not doing anything different with dedup right?